File: P06/PE_Problem06.js
/*
	Project Euler
	Vadim Brodsky
	Tuesday, September 4 2012
	Problem 06
*/
"use strict";
var timer = require("../timer.js");
timer.startTimer();

function sumOfSquares (range) {
	var sum = 0;
	for (var i = 1; i <= range; i++) {
		sum = sum + Math.pow(i, 2);
	}
	return sum;
}

function squareOfSum (range) {
	var sum = 0;
	for (var i = 1; i <= range; i++) {
		sum = sum + i;
	}
	return Math.pow(sum, 2);
}

function difference (range) {
	return squareOfSum(range) - sumOfSquares(range);
}

console.log(difference(100));
timer.endTimer();

File: P06/PE_Problem06.rb
require 'benchmark'

def sum_of_squares(range)
	return range.inject do |memo, n|
		memo + n ** 2
	end
end

def square_of_sum(range)
	sum = range.inject do |memo, n|
		memo + n
	end
	sum ** 2
end

def difference_v1(range)
	square_of_sum(range) - sum_of_squares(range)
end


# Code benchmark
Benchmark.bm(5) do |bm|
  bm.report('V1: ') do
	difference_v1(1..100)    
  end
end

puts difference_v1(1..100)